packages/ra-ui-materialui/src/input/ArrayInput/ArrayInput.tsx

```diff
@@ -126,11 +126,15 @@ export const ArrayInput = (props: ArrayInputProps) => {
     // We need to register the array itself as a field to enable validation at its level
     useEffect(() => {
         register(finalSource);
-        formGroups.registerField(finalSource, formGroupName);
+        formGroups &&
+            formGroupName != null &&
+            formGroups.registerField(finalSource, formGroupName);
 
         return () => {
             unregister(finalSource, { keepValue: true });
-            formGroups.unregisterField(finalSource, formGroupName);
+            formGroups &&
+                formGroupName != null &&
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** ‏ Same here
```suggestion
        formGroups &&
            finalSource &&
            formGroupName != null &&
```